File: Chapter-5-Classes-and-Objects/007-a-sample-program-to-introduce-arrays-of-objets.cpp
// a sample program to introduce arrays of objets
#include <bits/stdc++.h>

using namespace std;

class employee
{
    char name[30];
    float age;

public:
    void getData();
    void putData();
};

void employee::getData()
{
    cout << "Enter name : ";
    cin >> name;
    cout << "Enter age : ";
    cin >> age;
}

void employee::putData()
{
    cout << "Name : " << name << endl;
    cout << "Age : " << age << endl;
}

const int arraysize = 3;

int main()
{
    employee manager[arraysize];
    for (int i = 0; i < arraysize; i++)
    {
        cout << "\nDetails of manager #" << i + 1 << endl;
        manager[i].getData();
    }
    cout << endl;
    for (int i = 0; i < arraysize; i++)
    {
        cout << "\nManager #" << i + 1 << endl;
        manager[i].putData();
    }
}

// input
// Details of manager #1
// Enter name : Alice
// Enter age : 35

// Details of manager #2
// Enter name : Bob
// Enter age : 40

// Details of manager #3
// Enter name : Carol
// Enter age : 45

// output
// Manager #1
// Name : Alice
// Age : 35

// Manager #2
// Name : Bob
// Age : 40

// Manager #3
// Name : Carol
// Age : 45

File: Chapter-5-Classes-and-Objects/007-a-sample-program-to-introduce-arrays-of-objets.py
class Employee:
    def __init__(self):
        self.name = ''
        self.age = 0.0
    
    def get_data(self):
        self.name = input('Enter employee name: ')
        self.age = float(input('Enter employee age: '))
    
    def put_data(self):
        print("Name: ", self.name)
        print("Age: ", self.age)
    
array_size = 2

if __name__ == '__main__':
    managers = [
        Employee()
        for _ in range(array_size)
    ]

    for i in range(array_size):
        print(f'\nDetails of Manager {i+1}:')
        managers[i].get_data()
    for i in range(array_size):
        print(f'\nManager {i+1}:')
        managers[i].put_data()